source/Mission.cpp

```diff
@@ -1189,7 +1196,7 @@ Mission Mission::Instantiate(const PlayerInfo &player, const shared_ptr<Ship> &b
 	{
 		const Trade::Commodity *commodity = nullptr;
 		if(cargo == "random")
-			commodity = PickCommodity(*source, *result.destination->GetSystem());
+			commodity = PickCommodity(*sourceSystem, *result.destination->GetSystem());
```

Comment:
I am not sure why this was changed?
Do note that I am all for just upgrading things in general but I think that this change is unrelated to this PR and not needed. If I remember correclty `source` is used in so many files that changing it only in a few would not make sense, and it is unrelated to this PR.

Reply:
This one was confusing because `Mission::source` exists and is a Planet instead of a System. The method here defines its own `System *source`. I was meddling with that code since I used that variable.